File: C++/Make Binary Tree From Linked List.cpp
void convert(Node *head, TreeNode *&root) {
    vector<int> list;
    Node* ptr = head;
    while(ptr != NULL) {
        list.push_back(ptr -> data);
        ptr = ptr -> next;
    }
    queue<TreeNode*> q;
    root = new TreeNode(list[0]);
    q.push(root);
    int itr = 0;
    while(!q.empty()) {
        int n = q.size();
        for(int i=0; i<n; i++) {
            auto node = q.front();
            q.pop();
            if((2*itr+1) < list.size()) {
                node -> left = new TreeNode(list[2*itr+1]);
                q.push(node -> left);
            }
            if((2*itr+2) < list.size()) {
                node -> right = new TreeNode(list[2*itr+2]);
                q.push(node -> right);
            }
            itr++;
        }
    }
    return;
}

File: JAVA/Make Binary Tree From Linked List.java
class GfG 
{
    public static Tree convert(Node head, Tree node) {
        ArrayList<Tree> V=new ArrayList<>();
        Node temp=head;
        while(temp!=null){
            Tree treenode=new Tree(temp.data);
            V.add(treenode);
            temp=temp.next;
        }
        int n=V.size();
        for(int i=0;i<n/2;i++){
            if((2*i+1)<n){
               V.get(i).left=V.get(2*i+1);
            }
            if((2*i+2)<n){
                V.get(i).right=V.get(2*i+2); 
            }
        }
        return V.get(0);
    }
}  
